xapian-core/cluster/cluster.cc

```diff
@@ -246,12 +236,15 @@ Point::initialize(TermListGroup &tlg, const Document &doc_) {
     for (TermIterator it = doc.termlist_begin(); it != doc.termlist_end(); ++it) {
 	doccount wdf = it.get_wdf();
 	string term = *it;
-	if (wdf < 1 || term[0]=='Z')
+	double termfreq = tlg.get_termfreq(term);
+	/** If term is not a stemmed term or the term indexes only one
+	 *  one document, do not compute TF-IDF scores for them
+	 */
+	if (wdf < 1 || term[0] != 'Z' || termfreq <= 1)
```

Comment:
Well, it would be up to the user what subclass of `ExpandDecider` (or a subclass of some new functor class if `ExpandDecider` isn't actually suitable) they passed, though perhaps there would be a default one.

I'd be OK to keep the hard-coded criteria here now and get it merged with a plan to address that separately.  I'm not sure about punting on everything James says here.  I'll try to dig in and understand what's being discussed in more detail.